matrix_client/api.py

```diff
@@ -258,10 +258,13 @@ def redact_event(self, room_id, event_id, reason, txn_id=None, timestamp=None):
         path = '/rooms/%s/redact/%s/%s' % (
             room_id, event_id, txn_id
         )
+        content = {}
+        if reason:
```

Comment:
Can actually just go `content["reason"] = reason` without guarding it with the if. Requests is smart enough to deal with `None` values. Just FYI, not blocking merge.